third_party/blink/renderer/modules/cobalt/h5vcc_metrics/h_5_vcc_metrics.cc

```diff
@@ -131,15 +161,14 @@ void H5vccMetrics::EnsureReceiverIsBound() {
       remote_h5vcc_metrics_.BindNewPipeAndPassReceiver(task_runner));
   remote_h5vcc_metrics_.set_disconnect_handler(WTF::BindOnce(
       &H5vccMetrics::OnCloseConnection, WrapWeakPersistent(this)));
-  remote_h5vcc_metrics_->AddListener(
-      receiver_.BindNewPipeAndPassRemote(task_runner));
 }
 
 void H5vccMetrics::OnCloseConnection() {
   remote_h5vcc_metrics_.reset();
   receiver_.reset();
 
-  // TODO: Keep track of in-flight Promises and reject them here.
+  // TODO(cobalt, b/372559349): Keep track of in-flight Promises and reject them
```

Comment:
@yell0wd0g , can you give me some more context on the issue here and what's needed for a fix?

Reply:
If the remote Mojo endpoint goes away (and we know that via the `set_disconnect_handler(...H5vccMetrics::OnCloseConnection())`, we need to reject all promises that were created and are pending on the `WrapPersistent(resolver)`s passed to Mojo; these are not rejected by default. 

An example is WebHID request-device calls [here](https://source.chromium.org/chromium/chromium/src/+/main:third_party/blink/renderer/modules/hid/hid.h;l=114;drc=08bac9d4c75f023df86a07be1986884f760dab94).

Reply:
K. I'll follow-up on this one.